Polygon_mesh_processing/include/CGAL/Polygon_mesh_processing/extrude.h

```diff
@@ -0,0 +1,323 @@
+ * surface mesh with a bottom and top part, both having the same graph combinatorics as `input` (except
+ * that the orientation of the faces of the bottom part is reversed). The bottom and the top parts are
+ * connected by a triangle strip between boundary cycles. The coordinates of the points associated to the
+ * vertices of the bottom and top part are first initialized to the same value as the corresponding
+ * vertices of `input`. Then for each vertex, a call to `bot` and `top` is done for the vertices of the
+ * bottom part and the top part, respectively.
+ * \attention `output` may be self intersecting.
+ * @tparam InputMesh a model of `FaceListGraph`
+ * @tparam OutputMesh a model of `FaceListGraph` and `MutableFaceGraph`
+ * @tparam NamedParameters1 a sequence of \ref pmp_namedparameters "Named Parameters" for `InputMesh`
+ * @tparam NamedParameters2 a sequence of \ref pmp_namedparameters "Named Parameters" for `OutputMesh`
+ * @tparam BottomFunctor a functor providing
+ * \code {.cpp}
+ * void operator()`(boost::graph_traits<InputMesh>::vertex_descriptor input_v,boost::graph_traits<OutputMesh>::vertex_descriptor output_v)
+ * \endcode
+ *  where `output_v` is the copy of `input_v` from `input` into the bottom part of `output`.
+ *
+ * @tparam TopFunctor a functor providing a similar `operator()` as `BottomFunctor`.
+ * @param input an open surface mesh to extrude.
+ * @param output a surface mesh that will contain the result of the extrusion.
+ * @param bot functor that will transform all points copied from
+ * `input` in order to shape the bottom part of the extrusion.
+ * @param top functor that will transform all points copied from
+ * `input` in order to shape the top part of the extrusion.
+ * @param np_in an optional sequence of \ref pmp_namedparameters "Named Parameters" among the ones listed below
+ *
+ * \cgalNamedParamsBegin
+ *    \cgalParamBegin{vertex_point_map}
+ *    the property map that contains the points associated to the vertices of `input`.
+ * If this parameter is omitted, an internal property map for `CGAL::vertex_point_t`
+ * should be available for the vertices of `input` \cgalParamEnd
+ * \cgalNamedParamsEnd
+ *
+ * * @param np_out an optional sequence of \ref pmp_namedparameters "Named Parameters" among the ones listed below
+ *
+ * \cgalNamedParamsBegin
+ *    \cgalParamBegin{vertex_point_map}
+ *    the property map that will contain the points associated to the vertices of `output`.
+ * If this parameter is omitted, an internal property map for `CGAL::vertex_point_t`
+ * should be available for the vertices of `output` \cgalParamEnd
+ * \cgalNamedParamsEnd
+ */
+template <class InputMesh,
+          class OutputMesh,
+          class BottomFunctor,
+          class TopFunctor,
+          class NamedParameters1,
+          class NamedParameters2
+          >
+void extrude_mesh(const InputMesh& input,
+                  OutputMesh& output,
+                  BottomFunctor& bot,
```

Comment:
missing const I think or we should take them by copy. Same remark for the overload without nps.